src/scenario/message_media_text_blob.c

```diff
@@ -0,0 +1,164 @@
+#include "message_media_text_blob.h"
+
+#include "core/log.h"
+#include "core/string.h"
+
+#include <stdlib.h>
+#include <string.h>
+
+#define MESSAGE_MEDIA_TEXT_BLOB_INITIAL_SIZE (100 * 1024)
+#define MESSAGE_MEDIA_TEXT_BLOB_SIZE_INCREASE_OVERSHOOT (10 * 1024)
+#define MESSAGE_MEDIA_TEXT_BLOB_INITIAL_ENTRIES 100
+#define MESSAGE_MEDIA_TEXT_BLOB_ENTRIES_INCREASE_OVERSHOOT 10
+
+static message_media_text_blob_t message_media_text_blob;
+
+static void resize_text_blob(int needed_space)
```

Comment:
And now I cannot replicate it again.

Maybe just take a look at the code and let me know. I'll continue testing to see if I can get the graphical glitch again.

Reply:
I did more testing last night and couldn't get the glitch again.
I'm starting to think it might have instead been changes on the other branch (that I worked on at the same time) for asset packing rework that mucked with the assets that I hadn't cleaned properly afterwards.

If you take a look at the code and seem happy with how it is done, then I'll apply this to the new PR.

Reply:
The dynamic code looks okay, just the issues I mention below.